Standards v1

lietu opened the pull request.

RomanSavran committed:
Ontology v1

- Added data types and restrictions
- Added properties description per type

RomanSavran committed:
Updates on v1 feedback

- Fixed wrong encoding
- Fixed locale strings (adding country part)
- Fixed subclassof pointing to PoT ontology
- Technical classes like SupportedAttribute, SupportedClass, SupportedType do not appear as context or class definition
- Updated links for PoT and DLI onotlogies
- Link fields updated to contain 'from' and 'to'
- Language indexing is applied
- Storey renamed to Story according to US English spelling
- Terms and external ontologies usage reorganized
- Added possibility to change property labels depending on class where property is used

RomanSavran committed:
Attributes refactoring + Remove v0

Changes
- v0 version removed
- DLI attributes added including 4 metadata properties
- Property names refactored/root level added

Known bugs:
- name field missing in context
- DataProduct class do not contain properties in current requirements
- metadata date fields for some reason have "string" type in requirements, need to recheck with Henna
- incorrect title is generated in ClassDefinition files for dli properties
- some dli classes contexts might be missing (like App) because requirements of dli class inclusion in PoT are outdated
- adding trailing slash to class IRIs lately might cause inconsistency in URLs after deployment
- Person class contains duplicated fields because of duplicated
properties in DLI and PoT ontologies
- properties refactoring still pending from requirements perspective, so some names may still be unclear

RomanSavran committed:
Attributes refactoring + Remove v0

Changes
- v0 version removed
- DLI attributes added including 4 metadata properties
- Property names refactored/root level added

Known bugs:
- name field missing in context
- DataProduct class do not contain properties in current requirements
- metadata date fields for some reason have "string" type in requirements, need to recheck with Henna
- incorrect title is generated in ClassDefinition files for dli properties
- some dli classes contexts might be missing (like App) because requirements of dli class inclusion in PoT are outdated
- adding trailing slash to class IRIs lately might cause inconsistency in URLs after deployment

RomanSavran committed:
Merge branch 'standards-v1' of https://github.com/PlatformOfTrust/standards into standards-v1

RomanSavran committed:
New property and class structure

- Updated DLI class structures adding all of them to contexts
- PoT properties updated after review
- Duplicated Person fields removed
- Vocabulary file structure changes
- Identity and Links root fields added
- Translations consistency fixed
- Properties hierarchy implemented
- Titles for DLI fields fixed

RomanSavran committed:
Readme update

- Updated readme file
- Add Identity context to maintain structure

RomanSavran committed:
Small readme formatting update

RomanSavran committed:
Create pot.jsonld

Upload root ontology

RomanSavran committed:
Fix deployment

RomanSavran committed:
Update built.sh

Fix ontospy parameters

RomanSavran committed:
Update pot.jsonld

- Nice json format
- Deployment testing

RomanSavran committed:
Update build.sh

- Removed input, since it is added as parameter

RomanSavran committed:
Update build.sh

RomanSavran committed:
Update build.sh

RomanSavran committed:
Update setup.sh

lietu commented on build.sh:
```diff
@@ -3,8 +3,8 @@
-readonly WORKDIR="/src"
-readonly ONT_FILE="/src/ontologies/pot.jsonld"
+readonly WORKDIR="/src/v1"
+readonly ONT_FILE="${WORKDIR}/ontology/pot.jsonld"
```
`ontology` -> `Ontology`

RomanSavran committed:
Update setup.sh

lietu commented on setup.sh:
```diff
@@ -1,5 +1,5 @@
-pip install --no-cache-dir ontospy==1.9.6
-pip install --no-cache-dir ontodocs==1.2.2
+pip install --no-cache-dir ontospy==1.9.8.2
+pip install --no-cache-dir django
```
Excuse me?

lietu reviewed (commented):
Need changes

RomanSavran committed:
Update pot.jsonld

RomanSavran committed:
Update build.sh

soderluk commented on v1/ClassDefinitions/Identity/Digital.jsonld:
```diff
@@ -0,0 +1,221 @@
+            "@prefix": true
+        },
+        "dli": {
+            "@id": "https://standards.lifeengine.io/ontologies/dli.jsonld#",
```
Needs to be updated to the correct new standard URL, e.g. `standards.lifeengine.io/v1/Ontology/dli.jsonld`.

soderluk replied:
Why was this resolved? The URL still is missing the `v1` from the URL.

RomanSavran replied:
Hi, this file is removed in later commits since it belongs to DLI ontology. Let me check if this was removed from PoT

RomanSavran replied:
File was removed in later commits. Files left should contain correct url

soderluk commented on v1/ClassDefinitions/Identity/Digital.jsonld:
```diff
@@ -0,0 +1,221 @@
+                    "xsd:base": "xsd:string"
+                }
+            },
+            "id": {
```
`id` should not be specified in any of the contexts/class definitions. This is something that the API will create, not something that a user can specify.

soderluk replied:
ID is still here.

soderluk commented on v1/ClassDefinitions/Identity/Digital.jsonld:
```diff
@@ -0,0 +1,221 @@
+                    "xsd:base": "xsd:string"
+                }
+            },
+            "inLinks": {
```
`inLinks` and `outLinks` are not something that a user can specify. It's a list of links that the API figures out. The `inLinks` and `outLinks` should not be specified.

soderluk commented on v1/ClassDefinitions/Identity/Digital.jsonld:
```diff
@@ -0,0 +1,221 @@
+                    "xsd:string"
+                ]
+            },
+            "type": {
```
The type is determined by the context, and should not be something a user can specify.

lietu commented on build.sh:
```diff
@@ -3,8 +3,8 @@
 readonly CNAME_ADDRESS="standards.oftrust.net"
-readonly WORKDIR="/src"
-readonly ONT_FILE="/src/ontologies/pot.jsonld"
+readonly WORKDIR="/src/v1"
```
You can't just discard `v1` when deploying the `Context`, `Vocabulary` and `ClassDefinitions` -files.

You probably also should not do that for the files generated by `ontospy`

The context etc files must end up in `https://standards.oftrust.net/v1/Context/...` and I would find it really weird if we only had one HTML generated which is some random version of the standards instead of the `v1`/`v1.1`/`v2` that you're actually interested in

lietu replied:
https://help.github.com/en/articles/redirects-on-github-pages We'll likely want to set up a redirect like this from `/index.html` that ends up in the GitHub pages that then points to `/v1/index.html`

RomanSavran replied:
Added Jekyll configuration

RomanSavran committed:
Update build.sh

Corrected Artifacts folder

RomanSavran committed:
Update build.sh

RomanSavran committed:
Update build.sh

RomanSavran committed:
Update HTML generation logic

- Fixing bug with missing /v1/ folder

soderluk commented on v1/ClassDefinitions/Identity/Digital/Link/OwnerOf.jsonld:
```diff
@@ -0,0 +1,312 @@
+                "@type": "pot:SupportedAttribute",
+                "subPropertyOf": "pot:identifier",
+                "pot:title": "Identity to",
+                "pot:required": false,
```
The `from` and `to` identities are always required. Otherwise you do not have a link between identities.
Should these even be in the class definition, I don't know, maybe not, since it's something that the API handles as well.

soderluk commented on v1/Context/Identity/Thing/Device/Sensor.jsonld:
```diff
@@ -0,0 +1,140 @@
+            "@id": "pot:lifeCycle/status",
+            "@nest": "data"
+        },
+        "context": {
```
No need to define the context itself in the context.

RomanSavran committed:
Redirect

Adding redirect from root / to /v1/index.html

RomanSavran committed:
Update build.sh

Fix of wrong copy-past example from internet

RomanSavran committed:
Update build.sh

No comments

soderluk reviewed (changes requested):
Please check the reviews.

lietu commented on .travis.yml:
```diff
@@ -1,3 +1,6 @@
+plugins:
```
This is definitely not the right file for this. Travis has nothing to do with Jekyll.

RomanSavran committed:
Updated on DLI-PoT separation

- DLI classes removed from PoT contexts
- Links separated from Identty
- Updated pot.jsonld header

RomanSavran committed:
Remove redirect

Remove redirect due to incorrect setup

RomanSavran committed:
Required fields update

from/to required fields update

RomanSavran committed:
New fields

- Add readonly field to ClassDefinition files
- Add readonly and required to core ontology
- Fix labels duplication during HTML generation

RomanSavran committed:
Vocabulary fix

RomanSavran committed:
Vocabulary fix

Wired commit required to overcome Windows case-insensitivity

RomanSavran committed:
Jekyll Redirect

RomanSavran commented:
I noticed major bug with required and readonly properties, they were incorrectly inherited from DLI, working on fix

RomanSavran committed:
Update requirements of fields

- DLI field requirements and readonly signs inherited incorrectly - fixed
- Change dli:name to required
- Change dli:data to required because of dli:name requirement
